### PR TITLE
Add datepicker using jquery-ui

### PR DESCRIPTION
**Pull request by lukasz-kaniowski**

### CODE REVIEW

**Comment by lukasz-kaniowski** [comment]
Currently we need to load `jquery` in browsify mode as well as in browser directly. `selection-buttons.js` requires jquery to work so we have a *chicken and egg* situation.
Overall seems that we have at least **89KB** of unnecessary js loaded. 
gov_uk_-_the_best_place_to_find_government_services_and_information

I would probably leave it as is for now. We could try to optimize this later if it will be required.

**Comment by lukasz-kaniowski** [comment]
On mobile dates are by default in `dd.mm.yyyy` format

**Comment by gjvis** [comment]
on mobile the dates will follow the regional settings of the device right? so this might be simpler if we progressively enhance from three separate text-inputs to the single datepicker?

**Comment by gjvis** [comment]
Yeah just checked and the presentation format (on IOS 9.3.2 at least) changes with the decide's regional settings. The wire format *should* be consistently yyyy-mm-dd though.

**Comment by lukasz-kaniowski** [comment]
I've loosen up validation for deadline and allowed `yyyy-mm-dd`.

File: routes/jobs.js [hunk 1]
```diff
@@ -6,7 +6,19 @@ const progression = require('../lib/progression');
...
+function validateDeadline(deadline, req) {
+  if (deadline) {
+    if (deadline.includes('-')) {
+      req.checkBody('deadline', 'Deadline should be in "dd/mm/yyyy" format').isDate();
+    } else {
+      req.checkBody('deadline', 'Deadline should be in "dd/mm/yyyy" format')
+        .matches(/^(0?[1-9]|[12][0-9]|3[01])[\/](0?[1-9]|1[012])[\/]\d{4}$/);
+    }
+  }
 }
```
Comment (gjvis) [inline routes/jobs.js]: How do you feel about guard clauses?

e.g.
```javascript
function validateDeadline(deadline, req) {
  if (!deadline) { return; }

  if (deadline.includes('-')) {
    req.checkBody('deadline', 'Deadline should be in "dd/mm/yyyy" format').isDate();
  } else {
    req.checkBody('deadline', 'Deadline should be in "dd/mm/yyyy" format')
      .matches(/^(0?[1-9]|[12][0-9]|3[01])[\/](0?[1-9]|1[012])[\/]\d{4}$/);
  }
}
```

File: routes/jobs.js [hunk 2]
```diff
@@ -6,7 +6,19 @@ const progression = require('../lib/progression');
...
 function parseDeadline(deadlineString) {
-  return deadlineString ? moment(deadlineString, 'DD/MM/YYYY').format() : null;
+  const dateFormat = !deadlineString.includes('-') ? 'DD/MM/YYYY' : undefined;
+  return deadlineString ? moment(deadlineString, dateFormat).format() : null;
+}
+
+function validateDeadline(deadline, req) {
+  if (deadline) {
+    if (deadline.includes('-')) {
+      req.checkBody('deadline', 'Deadline should be in "dd/mm/yyyy" format').isDate();
```
Comment (gjvis) [inline routes/jobs.js]: Is this the correct error message?

File: test/integration/add-job-page.spec.js [hunk 3]
```diff
@@ -26,19 +26,25 @@ describe('Add a job page', () => {
...
+          .then(() => expect(addJobPage.getValidationError())
+            .to.equal('Deadline should be in "dd/mm/yyyy" format'))
+      );
+    });
+
+    [
+      { name: 'desktop format "dd/mm/yyyy"', date: '21/10/2016' },
+      { name: 'mobile format "yyyy-mm-dd"', date: '2016-12-21' },
+      { name: 'empty date', date: '' },
+    ].forEach(scenario => {
```
Comment (gjvis) [inline test/integration/add-job-page.spec.js]: <3

File: routes/jobs.js (hunk 2 above)
Comment (lukasz-kaniowski) [inline routes/jobs.js]: This one is tricky. It is correct browser -> on browser we want to push user to use 'DD/MM/YYYY'.
On mobile there will be native date support so this error should not be shown in theory.

Comment (gjvis) [inline routes/jobs.js]: Yeah, I can't think of anything better either

File: routes/jobs.js (hunk 1 above)
Comment (lukasz-kaniowski) [inline routes/jobs.js]: Looks nicer.

**Pull request merged**